Display shortcut hints for operators

BaoNTGH opened the pull request:
This is proposal for #95 

- Extract out all the operators' names into a separate module to be
  reused without introducing cycle dependencies. Also put all the names
  in one place
- Original thought was to have each operator defines its own keymap (makes
  more sense in term single responsibility) but due to the nature of key
  mapping where management/cross reference are more important, move them
  to a common place keymaps.py to be shared across.

Multiple shortcuts:
![image](https://user-images.githubusercontent.com/29285869/174170305-8775f0aa-9fbf-4f83-9833-d862bdf31175.png)

Single shortcut:
![image](https://user-images.githubusercontent.com/29285869/174170365-99522ec2-b91c-44ab-bfff-d080c3b5ae8e.png)

hlorus commented:
Really great addition and cleanup! I like the the hint to be displayed on top, right after the label.
What's currently a bit odd is that it's different from the workspacetool's tooltip, would be great to have them consistent, it should probably be possible to use the same function for both places.

BaoNTGH commented:
> Really great addition and cleanup! I like the the hint to be displayed on top, right after the label. What's currently a bit odd is that it's different from the workspacetool's tooltip, would be great to have them consistent, it should probably be possible to use the same function for both places.

Just to clarify that, you would prefer the single short cut screen above right? (It's possible to update the tool's hint to have similar as above single short cut.)

Also, for multiple shortcuts, would you be ok with above or also update to:
**Add Distance constraint (Alt + D, Alt + V, Alt + H)**

hlorus commented:
> > Really great addition and cleanup! I like the the hint to be displayed on top, right after the label. What's currently a bit odd is that it's different from the workspacetool's tooltip, would be great to have them consistent, it should probably be possible to use the same function for both places.
> 
> Just to clarify that, you would prefer the single short cut screen above right? (It's possible to update the tool's hint to have similar as above single short cut.)

Yeah, the same as in the screenshot you've posted.


> Also, for multiple shortcuts, would you be ok with above or also update to:
> **Add Distance constraint (Alt + D, Alt + V, Alt + H)**

Not sure about that, IMO both is okay. The only problem i see is that there's a maximum count of characters for the tooltip and it will get clipped if it's too long, so maybe better to display them inline.

BaoNTGH commented:
> > > Really great addition and cleanup! I like the the hint to be displayed on top, right after the label. What's currently a bit odd is that it's different from the workspacetool's tooltip, would be great to have them consistent, it should probably be possible to use the same function for both places.
> > 
> > Just to clarify that, you would prefer the single short cut screen above right? (It's possible to update the tool's hint to have similar as above single short cut.)
> 
> Yeah, the same as in the screenshot you've posted.
> 
> 
> > Also, for multiple shortcuts, would you be ok with above or also update to:
> > **Add Distance constraint (Alt + D, Alt + V, Alt + H)**
> 
> Not sure about that, IMO both is okay. The only problem i see is that there's a maximum count of characters for the tooltip and it will get clipped if it's too long, so maybe better to display them inline.
> 
> 

Ideally, I think one operator should do just one thing. It'll be less confusing. For example, direct distance, vertical distance and horizontal distance as there are actually different constraints IMO. It's clearer for user to use hot key as well. But I'll defer that for future discussion. Will update to be inlined for consistency for now

hlorus commented:
There is really just one distance constraint now with the option to be aligned or not. So this currently communicates the implementation quite well.
It might be possible to check through properties when searching for a keymap item and then expose the distance constraint with three buttons but not sure if it's really worth the effort...

BaoNTGH commented:
> There is really just one distance constraint now with the option to be aligned or not. So this currently communicates the implementation quite well. It might be possible to check through properties when searching for a keymap item and then expose the distance constraint with three buttons but not sure if it's really worth the effort...

Addressed all concerns. Took a chance to do much more cleaning up to reduce dependencies between modules.

![image](https://user-images.githubusercontent.com/29285869/174348413-b8e36653-d799-4f68-8edd-bf9bcdefac19.png)
![image](https://user-images.githubusercontent.com/29285869/174348455-166a7470-ebf6-429f-b1ed-c5e87eb7d0e1.png)

hlorus commented:
Great, the only thing that's kinda missing is support for global keymaps, see: 0b22b0ec69fe8bf82452efbe14d7d65c14d677b9.

hlorus commented:
As this PR is kinda getting difficult to review i'm gonna merge this already.

BaoNTGH commented:
> Great, the only thing that's kinda missing is support for global keymaps, see: 0b22b0ec69fe8bf82452efbe14d7d65c14d677b9.

Yup, that could come later. There are a little more clean up to centralize all the key map at one place
